solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/LTRScoringAlgorithm.java

```diff
@@ -169,5 +198,24 @@ public abstract Explanation explain(LeafReaderContext context, int doc,
   public String toString() {
     return  getClass().getSimpleName() + "(name="+getName()+")";
   }
+  
+  /**
+   * Goes through all the stored feature values, and calculates the normalized
+   * values for all the features that will be used for scoring.
+   */
+  private void normalizeFeatures(float[] modelFeatureValues) {
+    modelFeatureValuesNormalized = modelFeatureValues.clone();
+    for(int idx = 0; idx < modelFeatureValuesNormalized.length; ++idx) {
+      if (idx < norms.size()) {
+        modelFeatureValuesNormalized[idx] = 
+            norms.get(idx).normalize(modelFeatureValuesNormalized[idx]);
+      }
+    }
+  }
+  
+  public float[] getNormalizedFeatures(float[] modelFeatureValues) {
```

Comment:
Do we need this function?  Can the one above just return a modelFeatureValuesNormalized[], and then we can delete the member variable?